mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2087,18 +2176,41 @@ where
     }
 }
 
-pub fn check_all_utxo_inputs_signed_by_pub(
+pub fn check_all_utxo_inputs_signed_by_pub<T: UtxoCommonOps>(
+    coin: &T,
     tx: &UtxoTx,
     expected_pub: &[u8],
 ) -> Result<bool, MmError<ValidatePaymentError>> {
-    for input in &tx.inputs {
+    let expected_pub =
+        H264::from_slice(expected_pub).map_to_mm(|e| ValidatePaymentError::TxDeserializationError(e.to_string()))?;
+    for (idx, input) in tx.inputs.iter().enumerate() {
         let pubkey = if input.has_witness() {
             pubkey_from_witness_script(&input.script_witness).map_to_mm(ValidatePaymentError::TxDeserializationError)?
         } else {
             let script: Script = input.script_sig.clone().into();
+            if does_script_spend_p2pk(&script) {
+                let mut unsigned_tx: TransactionInputSigner = tx.clone().into();
+                // FIXME: For overwintered txs, we also need to set the input amount as it's used in sighash calcuations!
+                unsigned_tx.consensus_branch_id = coin.as_ref().conf.consensus_branch_id;
```

Comment:
to fix this, we need to query for the amount of this input. so one rpc request per p2pk input.

i think the efficient solution for this is to disable this check all together for p2pk spends on overwintered txs.

Reply:
I think you may just query the previous scriptPubKey for p2pk inputs.
This would be a one for all solution as you would not need to check the signature but just retrieve the pubkey from the previous scriptPubKey

Reply:
that's the same communication overhead, but i think is a neater solution.

we wouldn't wanna perform it though if the transaction/coin isn't overwintered, because we already have all the info needed to verify the taker is the owner of the utxos without the need for a network request.

Reply:
> we wouldn't wanna perform it though if the transaction/coin isn't overwintered, because we already have all the info needed to verify the taker is the owner of the utxos without the need for a network request.

I prefer to not use a network request when it's not needed.